cukes-rest/src/main/java/lv/ctco/cukesrest/internal/context/BaseContextHandler.java

```diff
@@ -9,10 +9,14 @@
 
     protected List<String> extractGroups(String pattern) {
         List<String> allMatches = new ArrayList<String>();
-        Matcher m = Pattern.compile(GROUP_PATTERN).matcher(pattern);
-        while (m.find()) {
-            allMatches.add(m.group(1));
+
+        if (pattern != null) {
```

Comment:
This was necessary for the `optional` argument in the following regex: `^response contains an array "(.+)" of size (>=|>|<=|<)?\s?(\d+)$`

I couldn't really follow the `ContextInflater`. Maybe you can propose and alternative but this is the stacktrace:

```
java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Matcher.java:1283)
	at java.util.regex.Matcher.reset(Matcher.java:309)
	at java.util.regex.Matcher.<init>(Matcher.java:229)
	at java.util.regex.Pattern.matcher(Pattern.java:1093)
	at lv.ctco.cukesrest.internal.context.BaseContextHandler.extractGroups(BaseContextHandler.java:13)
	at lv.ctco.cukesrest.internal.context.ContextInflater.inflate(ContextInflater.java:15)
	at lv.ctco.cukesrest.formatter.CukesRestJsonFormatter.match(CukesRestJsonFormatter.java:28)
	at sun.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at cucumber.runtime.Utils$1.call(Utils.java:37)
	at cucumber.runtime.Timeout.timeout(Timeout.java:13)
```

The specific code is here:
```java
        for (Argument argument : match.getArguments()) {
            String inflatedVal = contextInflater.inflate(argument.getVal());
            inflatedArguments.add(new Argument(argument.getOffset(), inflatedVal));
        }
```